obs2ioda-v2/src/f_c_string_t_mod.f90

```diff
@@ -0,0 +1,128 @@
+! Module providing functionality for handling a single Fortran string as a
+! C-compatible null-terminated string.
+module f_c_string_t_mod
+    use iso_c_binding, only : c_loc, c_ptr, c_null_char, &
+            c_char, c_f_pointer, c_null_ptr, c_size_t, c_associated, c_int, c_size_t
+    implicit none
+
+    public :: strlen
+
+    public :: f_c_string_t
+
+    interface
+        ! Declares an interface for the C function `strlen` to calculate the length
+        ! of a null-terminated C string.
+        !
+        ! Arguments:
+        ! - c_string: A C pointer to the null-terminated string whose length is to be calculated.
+        !
+        ! Returns:
+        ! - n: The length of the C string as an integer of kind `c_size_t`.
+        function c_strlen(c_string) bind(C, name = "strlen") result(n)
+            import :: c_ptr, c_size_t
+            type(c_ptr), value :: c_string
+            integer(c_size_t) :: n
+        end function c_strlen
+    end interface
+
+    ! Type to represent a single Fortran string that can be converted to/from
+    ! a C-compatible null-terminated string.
+    !
+    ! Fields:
+    ! - fc_string: Allocatable character string in C-compatible format
+    !   (null-terminated).
+    type :: f_c_string_t
+        character(len = :, kind = c_char), allocatable :: fc_string
+
+    contains
+        procedure :: to_c => to_c
+        procedure :: to_f => to_f
+        final :: cleanup
+    end type f_c_string_t
+
+contains
+    ! Computes the length of a null-terminated C string.
```

Comment:
It's personal preference, but I'd suggest a blank line after the `contains` statement.